assets/js/components/settings/SettingsCardKeyMetrics.js

```diff
@@ -69,43 +73,71 @@ export default function SettingsCardKeyMetrics() {
 	return (
 		<Layout title={ __( 'Key Metrics', 'google-site-kit' ) } header rounded>
 			<div className="googlesitekit-settings-module googlesitekit-settings-module--active googlesitekit-settings-user-input">
-				<SettingsKeyMetrics />
+				{ isUserInputCompleted && (
+					<Fragment>
+						<SettingsKeyMetrics loading={ loading } />
+
+						<Grid>
+							<Row>
+								<Cell size={ 12 }>
+									<UserInputPreview
+										settingsView
+										loading={ loading }
+									/>
+								</Cell>
+							</Row>
+						</Grid>
+					</Fragment>
+				) }
 
-				<Grid>
-					{ isUserInputCompleted && (
-						<Row>
-							<Cell size={ 12 }>
-								<UserInputPreview settingsView />
-							</Cell>
-						</Row>
-					) }
+				{ isUserInputCompleted === false && (
+					<Fragment>
+						<SettingsKeyMetrics />
```

Comment:
There's a divergence from the IB here, because we _don't_ want to show the "Display key metrics" toggle in the loading state in the case where `isUserInputCompleted` is false.

This is because we may not end up showing the toggle at all, if no key metrics have been chosen yet by the user. In this case it wouldn't make sense to initially show a loading placeholder for the toggle, only for it to then disappear once the key metrics state has been resolved.